gen/models/resource-properties/sns-subscription.json

```diff
@@ -0,0 +1,18 @@
+{
+  "Name": "SNSTopicSubscription",
```

Comment:
That seems find to me, nice pragmatic solution. I blame AWS for the name collision :)